packages/admin/cms-admin/src/generator/future/utils/deepKeyOf.ts

```diff
@@ -0,0 +1,15 @@
+// https://stackoverflow.com/a/58436959
+
+// export type DeepKeyOf<T> = T extends object
+//     ? {
+//           [K in keyof T]: `${Exclude<K, symbol>}${DeepKeyOf<T[K]> extends never ? "" : `.${DeepKeyOf<T[K]>}`}`;
+//       }[keyof T]
+//     : never;
+
+type Prev = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...0[]];
+type Join<K, P> = K extends string | number ? (P extends string | number ? `${K}${"" extends P ? "" : "."}${P}` : never) : never;
+export type DeepKeyOf<T, D extends number = 10> = [D] extends [never]
```

Comment:
Fancy. Why does the first, commented out, version not work?

Reply:
because it throws an error for a recursive structure like in the demo project. This is handled with Prev-Type.

I also tried to replace Prev and Join with the same-named Types of ts-toolbelt but they behave differently. And I'm actually not so experienced with typescript to find a library-supported solution in an reasonable amount of time.